scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadata.java

```diff
@@ -148,31 +218,56 @@ public Integer getId() {
 
       @Override
       public Set<Class<? extends ScriptMetadata>> getTypeClasses() {
-         return Util.<Class<? extends ScriptMetadata>>asSet(ScriptMetadata.class);
+         return Util.<Class<? extends ScriptMetadata>> asSet(ScriptMetadata.class);
       }
 
       @Override
       public void writeObject(ObjectOutput output, ScriptMetadata object) throws IOException {
-         output.writeInt(object.properties.size());
-         for(Entry<MetadataProperties, String> e : object.properties.entrySet()) {
-            output.writeUTF(e.getKey().name());
-            output.writeUTF(e.getValue());
-         }
+         output.writeUTF(object.name);
+         output.writeUTF(object.extension);
+         output.writeUTF(object.mode.name());
+         writeSet(output, object.parameters);
+
+         OptionalObjectInputOutput.writeOptional(output, object.language);
```

Comment:
For some reason I missed that, and a much better approach than my horrible one :)